File: back-end/ecommerce/src/main/java/org/generation/brazil/ecommerce/endereco/Endereco.java
package org.generation.brazil.ecommerce.endereco;


import lombok.Data;
import org.generation.brazil.ecommerce.cliente.Cliente;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Data
@Entity
public class Endereco {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_cliente")
    private Cliente cliente;

    @NotBlank
    private String logradouro;

    @NotBlank
    private String numero;

    @NotBlank
    private String cep;

    @NotBlank
    private String bairro;

    @NotBlank
    private String cidade;

    @NotBlank
    private String estado;

    private String complemento;

    private String referencia;
}

File: front-end/eco-mmerce/src/app/cliente/endereco.ts
import {Cliente} from './cliente';

export class Endereco {
  id: number;
  cliente: Cliente;
  logradouro: string;
  numero: string;
  cep: string;
  bairro: string;
  cidade: string;
  estado: string;
  complemento: string;
  referencia: string;

}
